### PR TITLE
make header file sample dependent in bcftools/annotate

### PR DESCRIPTION
**Pull request by nvnieuwk**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Review by matthdsm — APPROVED** [review]

File: tests/modules/nf-core/bcftools/annotate/nextflow.config [hunk 1]
```diff
@@ -1,6 +1,7 @@
 process {
     withName: 'BCFTOOLS_ANNOTATE_VCF' {
-        ext.args = "-x ID,INFO/DP,FORMAT/DP --output-type v"
+        ext.prefix = "test_vcf"
```
Comment (maxulysse) [inline tests/modules/nf-core/bcftools/annotate/nextflow.config:3]: why not keeping `test`?

File: tests/modules/nf-core/bcftools/annotate/test.yml [hunk 2]
```diff
@@ -1,23 +1,14 @@
...
+    - path: output/bcftools/test_vcf.vcf.gz
     - path: output/bcftools/versions.yml
 
 - name: bcftools annotate test_bcftools_annotate_out_bcf
-  command: nextflow run ./tests/modules/nf-core/bcftools/annotate -entry test_bcftools_annotate_out_bcf -c ./tests/config/nextflow.config
-  tags:
-    - bcftools/annotate
-    - bcftools
-  files:
-    - path: output/bcftools/test_ann.bcf
```
Comment (maxulysse) [inline tests/modules/nf-core/bcftools/annotate/test.yml:16]: How do you get this file name?

Comment (nvnieuwk) [reply tests/modules/nf-core/bcftools/annotate/test.yml:16]: It's in the config :)

**Comment by nvnieuwk** [comment]
Is this okay @maxulysse?

File: tests/modules/nf-core/bcftools/annotate/nextflow.config (hunk 1 above)
Comment (maxulysse) [reply tests/modules/nf-core/bcftools/annotate/nextflow.config:3]: of course :facepalm:

Comment (maxulysse) [inline tests/modules/nf-core/bcftools/annotate/nextflow.config:3]:
```suggestion
        ext.prefix = { "${meta.id}_vcf" }
```

File: tests/modules/nf-core/bcftools/annotate/test.yml (hunk 2 above)
Comment (maxulysse) [reply tests/modules/nf-core/bcftools/annotate/test.yml:16]: :facepalm: of course, it just wasn't modified, so I didn't saw it...

File: tests/modules/nf-core/bcftools/annotate/nextflow.config (hunk 1 above)
Comment (nvnieuwk) [reply tests/modules/nf-core/bcftools/annotate/nextflow.config]: Thanks for the review though 🙌

**Pull request merged**